src/test/java/ArrayIteratorTest.java

```diff
@@ -1,154 +1,163 @@
+            assertThrows(IllegalArgumentException.class, () -> new ArrayIterator<>(new Integer[]{}, 1));
+        }
 
-    @Test
-    void hasNext_endOfSizeOneCapacityTenArray_returnsFalse() {
-        Integer[] a = {10, null, null, null, null, null, null, null, null, null};
-        Iterator<Integer> it = new ArrayIterator<>(a, 1);
-        it.next();
-        assertFalse(it.hasNext());
-    }
 
-    @Test
-    void hasNext_endOfSizeFiveCapacityTenArray_returnsFalse() {
-        Integer[] a = {10, 11, 12, 13, 14, null, null, null, null, null};
-        Iterator<Integer> it = new ArrayIterator<>(a, 5);
-        for (int i = 0; i < 5; ++i) {
-            it.next();
+        @Test
+        void hasNext_capacityOne_returnsTrue() {
+            ArrayIterator<Integer> iterator = new ArrayIterator<>(new Integer[]{10}, 1);
+            assertTrue(iterator.hasNext());
         }
-        assertFalse(it.hasNext());
-    }
 
-    @Test
-    void hasNext_endOfFullCapacityTenArray_returnsFalse() {
-        Integer[] a = {10, 11, 12, 13, 14, 15, 16, 17, 18, 19};
-        Iterator<Integer> it = new ArrayIterator<>(a, 10);
-        for (int i = 0; i < 10; ++i) {
-            it.next();
+        @Test
+        void hasNext_capacityMany_returnsTrue() {
+            ArrayIterator<Integer> iterator = new ArrayIterator<>(new Integer[]{10, null, null, null, null}, 1);
+            assertTrue(iterator.hasNext());
         }
-        assertFalse(it.hasNext());
-    }
 
-    @Test
-    void next_capacityZeroArray_throwsException() {
-        Integer[] a = {};
-        Iterator<Integer> it = new ArrayIterator<>(a, 0);
-        assertThrows(NoSuchElementException.class, () -> it.next());
-    }
+        @Test
+        void next_capacityOne_returnsElement() {
+            ArrayIterator<Integer> iterator = new ArrayIterator<>(new Integer[]{10}, 1);
+            assertEquals(10, iterator.next());
+        }
 
-    @Test
-    void nest_endOfFullCapacityOneArray_throwsException() {
-        Integer[] a = {10};
-        Iterator<Integer> it = new ArrayIterator<>(a, 1);
-        it.next();
-        assertThrows(NoSuchElementException.class, () -> it.next());
-    }
+        @Test
+        void next_capacityMany_returnsElement() {
+            ArrayIterator<Integer> iterator = new ArrayIterator<>(new Integer[]{10, null, null, null, null}, 1);
+            assertEquals(10, iterator.next());
+        }
 
-    @Test
-    void next_sizeZeroCapacityTenArray_throwsException() {
-        Integer[] a = {null, null, null, null, null, null, null, null, null, null};
-        Iterator<Integer> it = new ArrayIterator<>(a, 0);
-        assertThrows(NoSuchElementException.class, () -> it.next());
-    }
+        @Test
+        void hasNext_postNextCapacityOne_returnsFalse() {
+            ArrayIterator<Integer> iterator = new ArrayIterator<>(new Integer[]{10}, 1);
+            iterator.next();
+            assertFalse(iterator.hasNext());
+        }
```

Comment:
You can merge a lot of tests here by collecting the iterator into a list, then asserting equality on the list. There's no "postNext" state to test.
```java
List<T> actual = new ArrayList<>();
classUnderTest.forEachRemaining(actual::add);
assertThat(actual).isEqualTo(List.of(...));
```

Reply:
The list comparison is a way to test iteration order. Why does it matter that you used that strategy in another test?

Reply:
Assuming the tests you were suggesting to merge with the use of the list equality strategy would be the 'next' returns elements in order & `next` after iterator is _done_. 

I figured, if the list comparison stratery failed, it could be because (a) there was an issue with the order/the number of elements, or (b) an issue with the iterator stopping where it should. The tests now have the test for the elements and then a test for what should happen when the iterator is at the end.

Reply:
I don't see any updates to the PR.